frontend/src/components/Board/Board.js

```diff
@@ -67,26 +66,54 @@ const Board = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [boardArray]);
 
-  const checkBoardCorrect = () => {
-    // Send this board to server
-    // const boardForServer = parseBoard(rows);
-    // Response from server
-    const boardCorrect = true;
-    if (boardCorrect) {
-      downloadNewBoard();
-    }
+  useEffect(() => {
+    ws = new WebSocket("ws:localhost/api/game");
+
+    ws.onmessage = function(event) {
+      let data2json = JSON.parse(event.data);
+      try {
+        data2json = JSON.parse(data2json);
+        console.log("Data from server: ", data2json);
+        if (event.data === "no_game") {
+          ws.close();
+          history.push("/lobby");
+        }
+        if (data2json.timeLeft) {
+          setTimeLeft(Math.round(data2json.timeLeft));
+        }
+        if (data2json.boardSolved) {
+          downloadNewBoard();
+        }
+        if (data2json.winner) {
+          alert("You won!");
+        }
+        if (data2json.timeout) {
+          alert("You lose");
+        }
+      } catch (e) {
+        console.log(e, "error");
+      }
+    }; // eslint-disable-next-line
```

Comment:
You should not omit dependencies, it will lead to bugs. Move the function definitions to the `useEffect` body and add dependencies as needed.